### PR TITLE
added template query

### PR DESCRIPTION
**Pull request by einorler**
Part of #97

### CODE REVIEW

File: src/Query/TemplateQuery.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        return 'template';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toArray()
+    {
+        $output = [
```
Comment (saimaz) [inline src/Query/TemplateQuery.php]: You are missing parameters trait and process function call.

File: src/Query/TemplateQuery.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            'inline' => $this->inline,
```
Comment (saimaz) [inline src/Query/TemplateQuery.php]: Missing `file` argument.

File: src/Query/TemplateQuery.php (hunk 1 above)
Comment (saimaz) [inline src/Query/TemplateQuery.php]: $output should be filtered.

File: src/Query/TemplateQuery.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    /**
+     * @var array
+     */
+    private $params;
+
+    /**
+     * @param string $inline A template of the query
+     * @param array  $params Parameters to insert into template
+     */
+    public function __construct($inline, array $params)
```
Comment (saimaz) [inline src/Query/TemplateQuery.php]: Missing `file`. This is kind of difficult situation, which one we should leave. IMO there should be both. Like this:

```php
public function __construct($file = null, $inline = null, array $params = [])
```

File: tests/Query/TemplateQueryTest.php [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+ * Unit test for Template.
+ */
+class TemplateQueryTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * Tests toArray() method with inline.
+     */
+    public function testToArrayInline()
+    {
+        $inline = '\"term\": {\"field\": \"{{query_string}}\"}';
```
Comment (saimaz) [inline tests/Query/TemplateQueryTest.php]: Text under single quote doesn't need escaping.

**Comment by einorler** [comment]
Removed escaping from single quoted strings in the test

**Pull request merged**